src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs

```diff
@@ -343,9 +345,25 @@ public ConfidentialClientApplicationBuilder WithTelemetryClient(params ITelemetr
                 Config.TelemetryClients = telemetryClients;
             }
 
+            TelemetryClientLogMsalVersion();
+
             return this;
         }
 
+        private void TelemetryClientLogMsalVersion()
+        {
+            MsalTelemetryEventDetails telemetryEventDetails = new MsalTelemetryEventDetails(TelemetryConstants.ConfigurationUpdateEventName);

```

Comment:
Do we need this event, if MSAL version is sent with every acquire token event?
Additionally, if we upload this event, when querying data how can we correlate MSAL version to the app, since we don't send like client ID?

Reply:
The consumer is very COGS sensitive, so they don't want extra data like client_id. We do have correlation id at the moment, but not sure if it will stay. This is a good question to follow up with the partners wanting to consume this.

Reply:
We don't log the client id but it is logged by default in the telemetry client as when a telemetry client object is created with a client id.

Reply:
Yes, correct. Client id is logged in all the tables by the telemetry client.